File: swift/Sources/SignalClient/zkgroup/ServerPublicParams.swift
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import Foundation
import SignalFfi

public class ServerPublicParams: ByteArray {

  public static let SIZE: Int = 225

  public required init(contents: [UInt8]) throws {
    try super.init(newContents: contents, expectedLength: ServerPublicParams.SIZE, unrecoverable: true)

    try withUnsafePointerToSerialized { contents in
      try checkError(signal_server_public_params_check_valid_contents(contents))
    }
  }

  public func verifySignature(message: [UInt8], notarySignature: NotarySignature) throws {
    try withUnsafePointerToSerialized { contents in
      try notarySignature.withUnsafePointerToSerialized { notarySignature in
        try checkError(signal_server_public_params_verify_signature(contents, message, message.count, notarySignature))
      }
    }
  }

}

File: java/java/src/main/java/org/signal/zkgroup/ServerPublicParams.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup;

import org.signal.zkgroup.internal.ByteArray;
import org.signal.client.internal.Native;

public final class ServerPublicParams extends ByteArray {
  public ServerPublicParams(byte[] contents)  {
    super(contents);
    Native.ServerPublicParams_CheckValidContents(contents);
  }

  public void verifySignature(byte[] message, NotarySignature notarySignature) throws VerificationFailedException {
    Native.ServerPublicParams_VerifySignature(contents, message, notarySignature.getInternalContentsForJNI());
  }

}
